File: 1.0-computer_science/1.0-cs.c/0.0-data-structure/callenges/1.0-arrInsert.js
function insertElement(arr, elem, index){
    var arrSize = 0;
    while (arr[arrSize] != undefined) arrSize += 1;
    for (; arrSize >= 0; arrSize--){
        if (arrSize === index){
            arr[arrSize] = elem;
            return;
        }else{
            arr[arrSize] = arr[arrSize - 1];
        }
    }
}

var array = [1, 2, 3, 4, 5];
var elem = 100;
var idx = 4;
console.log(array);
insertElement(array, elem, idx);
console.log(array);

File: 1.0-computer_science/1.0-cs.c/0.0-data-structure/callenges/1.0-arrinsert.c
#include <stdio.h>

void inserElement(int arr[], int elem, int idx){
    int size = 0;
    while (arr[size] != '\0') size++;
    for (; size >= 0; size--){
        if (size == idx){
            arr[size] = elem;
            return;
        }else {
            arr[size] = arr[size - 1];
        }
    }
}

int main(void){

    int array[6];
    array[0] = 1;
    array[1] = 2;
    array[2] = 3;
    array[3] = 4;
    array[4] = 5;
    array[5] = '\0';

    int size = 0;
    while (array[size] != '\0') size++;

    for (int i = 0; i < size; i++ )
        printf("%d ", array[i]);
    printf("\n");

    int idx = 5;
    int elem = 100;

    inserElement(array, elem, idx);

    for (int i = 0; i < size + 1; i++ )
        printf("%d ", array[i]);
    printf("\n");
}
